doc/opt-modes.md

```diff
@@ -0,0 +1,66 @@
+# Kubemacpool Opt-Modes
+
+Kubemacpool supports Opt-In and Opt-Out opt-modes, in order to control whether MAC Address will 
+be allocated to all VirtualMachines and Pods by default or not.
+- Opt-Out mode means that MAC Addresses for VirtualMachines/Pods are allocated by default, unless specifically chosen by adding a namespace label to exclude it.
+- Opt-In mode means that MAC Addresses for VirtualMachines/Pods are not allocated by default, unless specifically chosen by adding a namespace label to include it.
+
+The opt-mode can be set for Pods and VirtualMachines separately.
+
+**note:** The default opt-mode for [testing manifests](config/test/kubemacpool.yaml) is opt-out for VirtualMachines and Pods. 
+
+**note:** The default opt-mode for [release manifests](config/release/kubemacpool.yaml) is opt-out for VirtualMachines and opt-out for Pods.
+
+To set Kubemacpool to allocate all VirtualMachines and Pods by default (For more information on  [opt-out Mode](#How-to-opt-out-kubemacpool-MAC-assignment-for-a-namespace-in-Opt-out-mode)):
+```bash
+cp config/default/mutatepods_opt_out_patch.yaml config/test/mutatepods_opt_mode_patch.yaml
+cp config/default/mutatevirtualmachines_opt_out_patch.yaml config/test/mutatevirtualmachines_opt_mode_patch.yaml
+make generate-test
+```
+
+To set kubemacpool to not allocate all VirtualMachines and Pods by default (see [opt-in Mode](#How-to-opt-in-kubemacpool-MAC-assignment-for-a-namespace-in-Opt-in-mode))):
+```bash
+cp config/default/mutatepods_opt_in_patch.yaml config/test/mutatepods_opt_mode_patch.yaml
+cp config/default/mutatevirtualmachines_opt_in_patch.yaml config/test/mutatevirtualmachines_opt_mode_patch.yaml
+make generate-test
+```
+
+**note:** The User can of-course set Pods and VirtualMachines to separate opt-modes by sub-setting the above snippets. 
+
+## How to opt-out Kubemacpool MAC Address assignment for a namespace in Opt-out mode
```

Comment:
This is way too long for a title... What about changing these to just "Opting-in" and "Opting-out" - and explain details in the first paragraph.

Reply:
how about `Opt-in namespace when in Opt-in mode`?